src/Compilers/Core/Portable/Serialization/StreamObjectReader.cs

```diff
@@ -0,0 +1,1020 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    using EncodingKind = StreamObjectWriter.EncodingKind;
+    using Variant = StreamObjectWriter.Variant;
```

Comment:
seems odd ot have these defined in the writer type.  Why not have a Microsoft.CodeAnalysis.Serialization namespace and just put this stuff there?